### PR TITLE
fix matrix-free unit tests due to a change to DiagonalMatrix in Base …

### PR DESCRIPTION
**Pull request by davydden**
…class

while here, also use LaplaceOperator and add a few missing typedefs.

in reference to https://github.com/dealii/dealii/pull/3340

### CODE REVIEW

File: tests/matrix_free/parallel_multigrid_adaptive_02.cc [hunk 1]
```diff
@@ -50,107 +50,6 @@ std::ofstream logfile("output");
...
-        for (unsigned int i=0; i<phi.tensor_dofs_per_cell; ++i)
-          phi.begin_dof_values()[i] = local_diagonal_vector[i];
-        phi.distribute_local_to_global (dst);
-      }
-  }
-
-};
-
-
-
```
Comment (kronbichler) [inline tests/matrix_free/parallel_multigrid_adaptive_02.cc]: Removing this part makes sense, thanks!

File: include/deal.II/matrix_free/operators.h [hunk 2]
```diff
@@ -305,6 +305,15 @@ namespace MatrixFreeOperators
   class MassOperator : public Base<dim, Number>
   {
   public:
+    /**
+     * Number typedef.
+     */
+    typedef typename Base<dim,Number>::number_type number_type;
```
Comment (kronbichler) [inline include/deal.II/matrix_free/operators.h]: Do we have a policy somewhere on how we call these types? I'm used to naming them `value_type` in functions like this as that is the convention in the vector and other matrix classes. I'm fine with either variant but we need to do it consistently and I think consistency with other matrices is a good reason.

Comment (davydden) [inline include/deal.II/matrix_free/operators.h]: sure, how is it called in Trilinos and PETSc?

Comment (davydden) [inline include/deal.II/matrix_free/operators.h]: `typedef PetscScalar 	value_type`

Comment (davydden) [inline include/deal.II/matrix_free/operators.h]: `typedef TrilinosScalar 	value_type`

Comment (davydden) [inline include/deal.II/matrix_free/operators.h]: Switching to `value_type`...

**Comment by davydden** [comment]
@kronbichler I updated the PR.

**Pull request merged**